uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/infobar/Infobar.kt

```diff
@@ -0,0 +1,286 @@
+                defStyleRes
+            ).also {
+                setBackgroundResource(it.getResourceId(R.styleable.Infobar_android_background, 0))
+
+                setLeftButtonImageResource(
+                    it.getResourceId(R.styleable.Infobar_leftButtonIcon, 0),
+                    it.getResourceId(R.styleable.Infobar_leftButtonIconTint, 0)
+                )
+                setLeftButtonBackgroundResource(
+                    it.getResourceId(R.styleable.Infobar_leftButtonBackground, 0),
+                    it.getResourceId(R.styleable.Infobar_leftButtonBackgroundTint, 0)
+                )
+                setRightButtonImageResource(
+                    it.getResourceId(R.styleable.Infobar_rightButtonIcon, 0),
+                    it.getResourceId(R.styleable.Infobar_rightButtonIconTint, 0)
+                )
+                setRightButtonBackgroundResource(
+                    it.getResourceId(R.styleable.Infobar_rightButtonBackground, 0),
+                    it.getResourceId(R.styleable.Infobar_rightButtonBackgroundTint, 0)
+                )
+
+                elevation = it.getDimensionPixelSize(R.styleable.Infobar_android_elevation, 0).toFloat()
+                layoutMargin = it.getDimensionPixelSize(R.styleable.Infobar_android_layout_margin, 0)
+                layoutMarginTop = it.getDimensionPixelSize(R.styleable.Infobar_android_layout_marginTop, 0)
+                layoutMarginBottom = it.getDimensionPixelSize(R.styleable.Infobar_android_layout_marginBottom, 0)
+                layoutMarginStart = it.getDimensionPixelSize(R.styleable.Infobar_android_layout_marginStart, 0)
+                layoutMarginEnd = it.getDimensionPixelSize(R.styleable.Infobar_android_layout_marginEnd, 0)
+            }.recycle()
+        }
+    }
+
+    @CallSuper
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        with((layoutParams as MarginLayoutParams)) {
+            if (layoutMargin != 0) {
+                setMargins(layoutMargin, layoutMargin, layoutMargin, layoutMargin)
+            } else {
+                setMargins(layoutMarginStart, layoutMarginTop, layoutMarginEnd, layoutMarginBottom)
+            }
+        }
+    }
+
+    /**
+     * Sets the left button image resource and optionally an tint.
+     *
+     * @param imageResource [Int] to be used as left button image resource.
+     * @param tintColor [Int] optional parameter to be used as left button tint.
+     */
+    @JvmOverloads
+    fun setLeftButtonImageResource(@DrawableRes imageResource: Int, @ColorRes tintColor: Int = 0) {
+        if (imageResource != 0) binding.infobarLeftButton.setImageResource(imageResource)
+        if (tintColor != 0) setLeftButtonImageTint(tintColor)
+    }
+
+    /**
+     * Sets the left button image tint.
+     *
+     * @param tintColor [Int] left button tint.
+     */
+    fun setLeftButtonImageTint(@ColorRes tintColor: Int) {
+        if (tintColor != 0) binding.infobarLeftButton.tint(tintColor)
```

Comment:
kedze si to fixol v extensne, ma zmysel mat ten check v kazdej jednej metode?

Reply:
pravdu máš, nemá :(